lua/entities/cw_40mm_explosive/cl_init.lua

```diff
@@ -1,15 +1,15 @@
 include("shared.lua")
 
-function ENT:Initialize()
-	self.Emitter = ParticleEmitter(self:GetPos())
-	self.ParticleDelay = 0
-end
-
 function ENT:Draw()
 	self:DrawModel()
 end
 
 function ENT:Think()
+	if not self.Emitter then
+		self.Emitter = ParticleEmitter(self:GetPos())
```

Comment:
Is there any way for the emitter to be removed? Like is there a way for someone to conceptually spam create/destroy emitters?

Reply:
I dont know but Initialize might not always be called before the other hooks are called afaik.